### ISSUE TITLE
`useLocalStorage` and `useSessionStorage` is not supporting JS `Set` and `Map`

### ISSUE THREAD

**Issue by bayraak** [juliencrn/usehooks-ts#308]
`useLocalStorage` and `useSessionStorage` is not supporting JS `Set` and `Map` for `value` field.

## Steps to reproduce

```typescript
      const [oAuthErrors, setOAuthErrors] = useLocalStorage<Set<'Apple' | 'Facebook'>>(
        'oAuthErrors',
        new Set<'Apple' | 'Facebook'>()
    );
```

then try to set a new value somewhere else

```typescript
      setOAuthErrors(oAuthErrors.add('Facebook'));
```

## Expected Behaviour

It should work properly to serialize and deserialize `Set` and `Map` since when you give a type parameter to `useLocalStorage<Set>` is not complain about it.

## Actual Behaviour

The value set in `localStorage` or `sessionStorage` is `{}`

**Comment by juliencrn** [comment]
Should be fixed by #439. Thanks all 🙏

**Comment by juliencrn** [comment]
Should be fixed by #439, thanks for the feedback
